python/lsst/daf/persistence/butler.py

```diff
@@ -725,23 +727,29 @@ def _getOldButlerRepositoryCfg(repositoryArgs):
         cfg = RepositoryCfg.makeFromArgs(repositoryArgs)
         parent = PosixStorage.getParentSymlinkPath(repositoryArgs.cfgRoot)
         if parent:
-            cfg.addParents(parent)
+            parent = Butler._getOldButlerRepositoryCfg(RepositoryArgs(cfgRoot=parent, mode='r'))
+            if parent is not None:
+                cfg.addParents([parent])
         return cfg
 
     def _getRepositoryCfg(self, repositoryArgs):
         """Try to get a repository from the location described by cfgRoot.
 
         Parameters
         ----------
-        repositoryArgs : RepositoryArgs
-            Describes the location of a Repository.
+        repositoryArgs : RepositoryArgs, dict, or string
+            Provides arguments to load an existing repository (or repositories). String is assumed to be a URI
+            and is used as the cfgRoot (URI to the location of the cfg file).
 
         Returned
         --------
         (RepositoryCfg or None, bool)
             The RepositoryCfg, or None if one cannot be found, and True if the RepositoryCfg was created by
             reading an Old Butler repository, or False if it is a New Butler Repository.
         """
+        if not isinstance(repositoryArgs, RepositoryArgs):
+            repositoryArgs = RepositoryArgs(cfgRoot=repositoryArgs, mode='r')
```

Comment:
nice catch. copy-paste error on the docs. it should be `repositoryArgs : RepositoryArgs or string`, will fix.